Improved UI on client_id copy

samchan0221 opened the pull request:
ref #1110 
- [x] Added copy button
- [x] Added popup on copy
- [x] Remove old copy button logic
![image](https://user-images.githubusercontent.com/36394128/127976548-ad9af1cc-250d-4cce-bbef-b1b19f69ee13.png)

louischan-oursky commented on portal/src/graphql/portal/ApplicationsConfigurationScreen.tsx:
```diff
@@ -83,33 +93,87 @@ function makeOAuthClientListColumns(
+          iconProps={{ iconName: "Copy" }}
+          title="Copy"
+          ariaLabel="Copy"
+          styles={{ root: styles.copyButtonRoot }}
```
This object can be lifted as module-level constant.

louischan-oursky replied:
Can we add 1 more condition to dismiss the callout? The condition is after 0.5s when onMouseLeave fires on IconButton.

Can we tidy up the show/dismiss logic as a hook?

louischan-oursky commented on portal/src/graphql/portal/ApplicationsConfigurationScreen.tsx:
```diff
@@ -83,33 +93,87 @@ function makeOAuthClientListColumns(
+            directionalHint={DirectionalHint.topCenter}
+            onDismiss={handleCalloutDismiss}
+          >
+            <DelayedRender>
```
Can we avoid using DelayedRender?

samchan0221 commented:
@louischan-oursky 
I have committed the changes you requested
1. Moved icon props out as constant
2. Written a useDelayCallback hook
3. Removed the delay render wrapper

louischan-oursky commented on portal/src/graphql/portal/ApplicationsConfigurationScreen.tsx:
```diff
@@ -102,31 +101,48 @@ function makeOAuthClientListColumns(
 }
 
+function useDelayedAction(delayed: () => void): (delay: number) => void {
+  const [delay, setDelay] = useState<[number] | null>(null);
```
Add a comment to explain why `delay` is a tuple of 1 element.

louischan-oursky commented on portal/src/graphql/portal/ApplicationsConfigurationScreen.tsx:
```diff
@@ -101,24 +98,6 @@ function makeOAuthClientListColumns(
   ];
 }
 
-function useDelayedAction(delayed: () => void): (delay: number) => void {
```
The original version of the hook uses `useEffect` to correctly cleanup scheduled setTimeout when the component unmounts. I suggest we keep this version.

louischan-oursky commented on portal/src/graphql/portal/ApplicationsConfigurationScreen.module.scss:
```diff
@@ -55,3 +55,19 @@
 }
+
+.callout {
+  padding: 8px;
```
Use `styles` to custom the component itself. Use `className` to layout the component within its parent.

louischan-oursky commented on portal/src/graphql/portal/ApplicationsConfigurationScreen.module.scss:
```diff
@@ -55,3 +55,19 @@
+}
+
+.copyButtonRoot {
+  margin: 4px;
```
For example, this should be referred as `className={styles.copyButton}` (No suffix Root because we do not assume the internal HTML structure of the component)

louischan-oursky commented on portal/src/graphql/portal/ApplicationsConfigurationScreen.module.scss:
```diff
@@ -55,3 +55,19 @@
+
+.copyButtonRoot:active {
+  background-color: #c2c0be;
+}
```
Changing the hover and active color is of customizing the component. So this should be written using `styles` prop.
